p2p/simulations/http.go

```diff
@@ -423,6 +454,25 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func (s *Server) evalMsgFilter(activeMsgFilters map[string][]uint64, event *Event) bool {
+	matches := false
+	//client needs to at least pass "filterProtocol" to enable filtering
+	//if not, no messages will be delivered at all
+	if len(activeMsgFilters) > 0 {
```

Comment:
I'm pretty sure the Go implementation of a map will optimise lookups to return early if the map is empty. The only way to settle this would be for a benchmark, but I'm 99% confident it will show that this check is actually worse for performance when the map isn't empty (which with the given code path is always true).